langworld_db_data/adders/listed_value_adder.py

```diff
@@ -187,21 +191,25 @@ def _increment_ids_whose_indices_are_equal_or_greater_than_index_to_assign(
         index_to_assign: int,
     ) -> list[dict[str, str]]:
         """
-        Increases by 1 index of every value that will come after the value passed for insertion
+        Increases by 1 index of every value that will come after the value passed for insertion.
+        This helps to prepare rows for new value insertion.
         """
 
+        rows_with_incremented_indices = rows.copy()
         for value_index_and_line_number in value_indices_to_inventory_line_numbers:
             if value_index_and_line_number["index"] < index_to_assign:
                 continue
             row_where_id_must_be_incremented = value_index_and_line_number["line number"]
-            value_id_to_increment = rows[row_where_id_must_be_incremented]["id"]
+            value_id_to_increment = rows_with_incremented_indices[
+                row_where_id_must_be_incremented
+            ]["id"]
             components_of_value_id_to_increment = value_id_to_increment.split("-")
-            rows[row_where_id_must_be_incremented]["id"] = (
+            rows_with_incremented_indices[row_where_id_must_be_incremented]["id"] = (
                 f"{components_of_value_id_to_increment[0]}-{components_of_value_id_to_increment[1]}-"
                 f"{int(components_of_value_id_to_increment[2]) + 1}"
             )
 
-        return rows
+        return rows_with_incremented_indices
```

Comment:
```suggestion
        return tuple(rows_with_incremented_indices)
```
let's return a tuple to be consistent with the return type in a similar method?
If you agree, please change the type hint in method definition as well.